vm/src/memory/operand_stack.h

```diff
@@ -91,20 +94,34 @@ class OperandStack {
      */
     const char *pop_bytes(size_t n);
 
+    /**
+     * transfer to this operand stack the n first bytes of the given caller stack.
+     * The bytes are transferred without modifying the operands object refs vector
+     * in order to keep the information over contained references
+     * */
+    void transfer(OperandStack &caller_stack, size_t n);
+
     void push(const char *bytes, size_t size);
 
     template <typename T>
     void push(T t) {
         if (current_pos + sizeof(T) > stack_capacity) {
             throw StackOverflowError("exceeded stack capacity via operand stack");
         }
+
+        size_t false_bits_count = sizeof(T);
+        if constexpr (std::is_same_v<T, msh::obj *>) {
```

Comment:
This test is incorrect when using `const msh::obj*`, `msh::obj const*`, `msh::obj&`...